[21871] Ensure no word "Standalone" is appended to the standalone name

livecodepanos opened the pull request:
The `revStandalonePreBuild` command prepends the word "Standalone" to the standalone name, if there is already a file `pFolder & "/" & <standalone_name>`. When building for both `Linux` and `Linux 64`, the `revStandalonePreBuild` command is called twice, but both times with `pFolder`.

In this case `pFolder` is `/Users/panos/Downloads/<standalone_folder>/Linux`. So when `revStandalonePreBuild` is called for `Linux 64`, the file `pFolder & "/" & <standalone_name>` already exists (as it was created when building for `Linux`), so the word "Standalone" was prepended to the standalone name.

The fix is to call `revStandalonePreBuild` with `tDirectory` instead, as `tDirectory` is:
- `/Users/panos/Downloads/<standalone_folder>/Linux/`
- `/Users/panos/Downloads/<standalone_folder>/Linux x64/`

But we don't want the last `/`, so we actually need `char 1 to -2 of tDirectory`

runrevmark commented on ide-support/revsaveasstandalone.livecodescript:
```diff
@@ -928,14 +928,15 @@ command revSaveAsLinuxStandalone pStack, pRestoreFileName, pFolder, @xStandalone
       put tStandaloneSettingsA into sStandaloneSettingsA
       
       local tArch
+      set the itemDelimiter to comma
```
I suggest moving this after the extraction of tStackPath above so the change from the default item delimiter is localized to that bit of code.

runrevmark commented:
@livecodepanos : The PR message here is really useful - can you put it into the commit where the code is changed too?

livecodepanos committed:
Ensure no word "Standalone" is appended to the standalone name

The `revStandalonePreBuild` command prepends the word "Standalone"
to the standalone name, if there is already a file
`pFolder & "/" & <standalone_name>`. When building for both Linux
and Linux 64, the `revStandalonePreBuild` command is called twice,
but both times with `pFolder`.

In this case `pFolder` is `/Users/panos/Downloads/<standalone_folder>/Linux`.

So when `revStandalonePreBuild` is called for Linux 64, the file
`pFolder & "/" & <standalone_name>` already exists
(as it was created when building for Linux), so the word "Standalone"
was prepended to the standalone name.

The fix is to call `revStandalonePreBuild` with `tDirectory` instead, as
`tDirectory` is:

`/Users/panos/Downloads/<standalone_folder>/Linux/`
`/Users/panos/Downloads/<standalone_folder>/Linux x64/`

But we don't want the last /, so we actually need `char 1 to -2 of tDirectory`

livecodepanos commented:
@runrevmark sure, done :)

runrevmark commented:
@livecode-vulcan review ok ec5cfc6

livecode-vulcan commented:
💙 review by @runrevmark ok ec5cfc6a14c05b0315143393168fb86e811e9565

livecode-vulcan commented:
😎 test success ec5cfc6a14c05b0315143393168fb86e811e9565

* try-community-armv6-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-sdk26_ndk16r15/builds/329)
* try-community-armv7-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-armv7-android-sdk26_ndk16r15/builds/321)
* try-community-arm64-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-arm64-android-sdk26_ndk16r15/builds/321)
* try-community-x86-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86-android-sdk26_ndk16r15/builds/321)
* try-community-x86_64-android-sdk26_ndk16r15: [success](https://vulcan.livecode.com/builders/try-community-x86_64-android-sdk26_ndk16r15/builds/321)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1843)
* try-community-universal-ios-iphoneos12.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos12.1/builds/71)
* try-community-universal-ios-iphonesimulator12.1: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator12.1/builds/71)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1874)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1864)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1863)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1897)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1838)
